driver/src/main/java/org/neo4j/driver/internal/connector/socket/SocketResponseHandler.java

```diff
@@ -75,7 +74,7 @@ public void handleRecordMessage( Value[] fields )
             collector.record( fields );
         }
 
-        logger.debug( new RecordMessage( fields ).toString() );
+        logger.debug( "S: RecordMessage{%s}", Arrays.asList( fields ) );
```

Comment:
Same with these as above - we could move this to a separate MessageHandler, LoggingMessageHandler, and only include that if debug logging is enabled.